bin/install-wp-tests.sh

```diff
@@ -39,7 +39,23 @@ wp() {
 }
 
 wait_db() {
-	local MYSQLADMIN_FLAGS="--user=$DB_USER --password=$DB_PASS --host=$DB_HOST"
+	# parse DB_HOST for port or socket references
+	local PARTS=(${DB_HOST//\:/ })
+	local DB_HOSTNAME=${PARTS[0]};
+	local DB_SOCK_OR_PORT=${PARTS[1]};
+	local EXTRA=""
+
+	if ! [ -z $DB_HOSTNAME ] ; then
+		if [ $(echo $DB_SOCK_OR_PORT | grep -e '^[0-9]\{1,\}$') ]; then
+			EXTRA=" --host=$DB_HOSTNAME --port=$DB_SOCK_OR_PORT --protocol=tcp"
+		elif ! [ -z $DB_SOCK_OR_PORT ] ; then
+			EXTRA=" --socket=$DB_SOCK_OR_PORT"
+		elif ! [ -z $DB_HOSTNAME ] ; then
+			EXTRA=" --host=$DB_HOSTNAME --protocol=tcp"
+		fi
+	fi
+
+	local MYSQLADMIN_FLAGS="--user=$DB_USER --password=$DB_PASS $EXTRA"
```

Comment:
It looks like this code is duplicated here and in `install_db`. What do you think of extracting it to a new function, e.g. `get_db_connection_string`, and using that function in those places?

Reply:
Gimme a sec to try, I want to move at the very beginning of the script.

Reply:
The incorporated solution is proposed by @luizreis, thank you :+1: